### PR TITLE
Wire up get_dictionary_item to client FFI

### LINKED ISSUE #1886: Update dictionary functionality in client
The client needs to have the following changes related to the recent addition of support for getting a dictionary item:
* rename query_dictionary.rs to get_dictionary_item.rs
* consistently name functions and comments as "get_dictionary_item"
* add FFI function
* add integration test

### PR DESCRIPTION
**Pull request by darthsiroftardis**
CHANGELOG:

- Wired up `casper_get_dictionary_item()` to the client FFI
- Added tests for the function in integration tests
- Enforced consistency for `get_dictionary_item` name

Closes #1886

### CODE REVIEW

**Review by Fraser999 — APPROVED** [review]
LGTM!

File: client/lib/ffi.rs [hunk 1]
```diff
@@ -878,3 +912,98 @@ impl TryInto<super::SessionStrParams<'static>> for casper_session_params_t {
         })
     }
 }
+
+/// Container for `DictionaryItemStrParams` construction options.
+///
+/// See [DictionaryItemStrParams](super::DictionaryItemStrParams) for more info.
+#[allow(non_snake_case, missing_docs)]
```
Comment (rafal-ch) [inline client/lib/ffi.rs]: Rust seems to be content even without `non_snake_case` 🐍 
Can you double-check?

**Review by rafal-ch — APPROVED** [review]
LGTM 👍

File: client/lib/ffi.rs (hunk 1 above)
Comment (Fraser999) [reply client/lib/ffi.rs]: Well spotted.  Looks like `non_snake_case` is no longer required for types which are `#[repr(C)]`.  Please remove all the `allow(non_snake_case)` attributes in this file.

**Comment by darthsiroftardis** [comment]
bors try

File: client/lib/ffi.rs (hunk 1 above)
Comment (darthsiroftardis) [reply client/lib/ffi.rs]: Removed

**Pull request merged**